### ISSUE TITLE
Getting ext-pgsql into core

### ISSUE THREAD

**Issue by simonwelsh** [facebook/hhvm#7200]
This is mostly so I know what needs to be done once the main diff lands. Will also allow others to contribute if they want to :)

* [ ] land main diff (#6849)
* [ ] HHI file (not the Hack friendly version that's in hhvm-pgsql :( )
* [ ] Docblocks in HNI file
* [ ] Document PHP incompatibilities (mostly that the resource argument isn't optional. See the README in hhvm-pgsql for a more thorough list)
* [ ] Add `libpq5` as an install dependancy for pre-build packages
* [ ] Poke people to make sure the build machines have `libpq-dev` installed

**Comment by photodude** [comment]
Item 1 should be done now with the closure of #6849

As for PHP incompatibilities, (differences from the Zend module) there are about [24 missing functions when compared to the Zend pgsql module](https://github.com/PocketRent/hhvm-pgsql/tree/master#differences-from-zend) and 2 new functions that are not in the Zend module.

Personally, I would hope that `pg_set_client_encoding()` and pg_set_error_verbosity() could be added now that ext-pgsql has been merged into HHVM core (as for the other 22 missing functions when compared to the Zend pgsql module, I have less interest since our project doesn't use them).
